apps/meteor/server/models/raw/Rooms.ts

```diff
@@ -1970,4 +1970,80 @@ export class RoomsRaw extends BaseRaw<IRoom> implements IRoomsModel {
 
 		return this.updateOne(query, update);
 	}
+
+	addUserIdToE2EEQueueByRoomIds(roomIds: IRoom['_id'][], uid: IUser['_id']): Promise<Document | UpdateResult> {
+		const query: Filter<IRoom> = {
+			_id: {
+				$in: roomIds,
+			},
+			'usersWaitingForE2EKeys.userId': { $ne: uid },
+			encrypted: true,
+		};
+
+		const update: UpdateFilter<IRoom> = {
+			$push: {
```

Comment:
We should store 50 users at maximum, after that we need to rotate the users, removing the oldest ones to add the new ones.

Reply:
@KevLehman addToSet does not allow the usage of slice to limit the size of the array